debezium-core/src/main/java/io/debezium/relational/RelationalConnectorTask.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.relational;
+
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import org.apache.kafka.connect.errors.ConnectException;
+import org.apache.kafka.connect.source.SourceConnector;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.config.Configuration;
+import io.debezium.connector.base.ChangeEventQueue;
+import io.debezium.connector.common.BaseSourceTask;
+import io.debezium.jdbc.JdbcConnection;
+import io.debezium.pipeline.ChangeEventSourceCoordinator;
+import io.debezium.pipeline.DataChangeEvent;
+import io.debezium.pipeline.ErrorHandler;
+import io.debezium.pipeline.EventDispatcher;
+import io.debezium.pipeline.source.spi.ChangeEventSourceFactory;
+import io.debezium.pipeline.spi.OffsetContext;
+import io.debezium.pipeline.spi.OffsetContext.Loader;
+import io.debezium.schema.TopicSelector;
+import io.debezium.util.Clock;
+import io.debezium.util.LoggingContext;
+
+/**
+ * Base class for Debezium's CDC connector tasks for relational databases.
+ *
+ * @author Gunnar Morling
+ */
+public abstract class RelationalConnectorTask extends BaseSourceTask {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RelationalConnectorTask.class);
+
+    private static enum State {
+        RUNNING, STOPPED;
+    }
+
+    public interface StartingContext {
+
+        RelationalDatabaseConnectorConfig getConnectorConfig();
+        Class<? extends SourceConnector> getConnectorClass();
+        TopicSelector<TableId> getTopicSelector();
+        JdbcConnection getJdbcConnection();
+        RelationalDatabaseSchema getDatabaseSchema();
```

Comment:
`JdbcConnection` and `RelationalDatabaseSchema` should be generic types as the connectors extend them.

Reply:
I preferred simple casting in the sub-classes, as lots of type arguments get unwidely. But we can also revisit this later on.